### PR TITLE
CHAIN-475: testnet enrollment

### PR DESCRIPTION
**Pull request by bholzman**

### CODE REVIEW

**Review by bflood366 — APPROVED** [review]
looks great 👍

File: web-ui/src/components/Screens/HomeScreen/swap/ConnectWallet.tsx [hunk 1]
```diff
@@ -41,7 +41,6 @@ export function ConnectWallet({ onSwitchToChain }: Props) {
               : `Connect ${missingConnection}Wallet`
           }}
           onClick={() => {
-            console.log('ConnectWallet', 'onClick')
```
Comment (imykolenko) [inline web-ui/src/components/Screens/HomeScreen/swap/ConnectWallet.tsx:44]: 👌

**Review by imykolenko — APPROVED** [review]

File: backend/src/main/kotlin/xyz/funkybit/apps/api/model/Configuration.kt [hunk 2]
```diff
@@ -28,6 +31,11 @@ data class AccountConfigurationApiResponse(
     val newSymbols: List<SymbolInfo>,
     val role: Role,
     val authorizedAddresses: List<Address>,
+    val testnetChallengeStatus: TestnetChallengeStatus,
+    val testnetChallengeDepositSymbol: String?,
+    val testnetChallengeDepositContract: Address?,
```
Comment (antonyshchenko) [inline backend/src/main/kotlin/xyz/funkybit/apps/api/model/Configuration.kt:36]: Nit: seems like this three fields should be grouped together into a data class.

File: web-ui/src/components/Screens/HomeScreen/TestnetChallengeTab.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        setTestnetChallengeDepositContract(symbolContract)
+        if (symbol.chainId != evmConfig.state.chainId) {
+          setSwitchToEthChainId(symbol.chainId)
+        }
+        setShowTestnetChallengeDepositModal(true)
+      }
+    }
+  }, [accountConfigQuery, evmConfig.state.chainId, symbols])
+
+  const autoDepositTriggeredRef = useRef(false)
```
Comment (antonyshchenko) [inline web-ui/src/components/Screens/HomeScreen/TestnetChallengeTab.tsx:97]: Wondering why do we have to use refs here?

File: backend/src/main/kotlin/xyz/funkybit/apps/api/model/Configuration.kt (hunk 2 above)
Comment (bholzman) [reply backend/src/main/kotlin/xyz/funkybit/apps/api/model/Configuration.kt:36]: you're right, that would be cleaner

File: web-ui/src/components/Screens/HomeScreen/TestnetChallengeTab.tsx (hunk 3 above)
Comment (bholzman) [reply web-ui/src/components/Screens/HomeScreen/TestnetChallengeTab.tsx:97]: we don't, ended up this way after some debugging, I'm changing it to use a more standard `useState` in my current branch

**Pull request merged**